worker/uniter/util_test.go

```diff
@@ -283,7 +283,7 @@ func addControllerMachine(c *gc.C, st *state.State) {
 	// The AddControllerMachine call will update the API host ports
 	// to made-up addresses. We need valid addresses so that the uniter
 	// can download charms from the API server.
-	apiHostPorts, err := st.APIHostPorts()
+	apiHostPorts, err := st.APIHostPortsForClients()
```

Comment:
I'm not super concerned about this test case, but I wonder if Clients is correct here, or if it should actually be the agents.
Given the comment, it would seem it should be the agent addresses. Thoughts?

Reply:
I think it is correct.
It pulls out the current (full) set, adds the controller, then resets them.
As it is, without setting a management space, ForClients and ForAgents are equivalent.

SetAPIHostPorts populates 2 documents. For this case it is not important, but *if* you were using a management space *and* it was effectively filtering addresses, by using APIHostPortsForAgents you would have a transition like:

ForClients = All, ForAgents = Filtered -> ForClients = Filtered, ForAgents = Filtered.

Reply:
gotcha. I was actually reading it the wrong way around. I think you're correct.